python/cuml/ensemble/randomforestregressor.pyx

```diff
@@ -348,7 +357,10 @@ class RandomForestRegressor(Base):
         self.quantile_per_tree = quantile_per_tree
         self.n_streams = handle.getNumInternalStreams()
         self.seed = seed
-
+        tmpdir = tempfile.mkdtemp()
+        file_name = os.path.join(tmpdir, "model.buffer")
+        self.file_name = file_name
+        self.model_pbuf_bytes = []
```

Comment:
shouldn't this happen in the `getstate` and not here? Also you could generate a new temp file in `setstate` so that we don't need to store this file_name. It seems to me like a temp file should be temporary and not part of the state of the object